card/mod/05_email/set/all/follow.rb

```diff
@@ -118,16 +120,30 @@ def followed?
 
 
 def follow_rule_applies? user_id
-  if (follow_rule_card=rule_card(:follow, :user_id=>user_id))
-    follow_rule_card.item_cards.each do |item_card|
-      if item_card.respond_to?(:applies_to?) and item_card.applies_to? self, user_id
-         return item_card
+  follow_rule = rule :follow, :user_id=>user_id
+  if follow_rule.present?
+    follow_rule.split("\n").each do |value|
+      if test = FollowOption.test[ value.to_name.code ]
+        if test.call @follow_test_options.merge( :user_id => user_id )
+          return value.gsub( /[\[\]]/, '' )
+        end
+      else
+        Rails.logger.debug "didn't find follow test for #{ value }"
       end
     end
   end 
   return false
```

Comment:
this is the pivotal test code.  Much faster now.  no cards instantiated, no duplicate queries (eg in edited).  But more hard code than is optimal!